pkg/streamingpromql/operators/functions/functions.go

```diff
@@ -4,6 +4,20 @@ package functions
 
 var promQLNamesToFunctions = map[string]Function{}
 
+var FunctionsSupportingAnchoredMatrix = map[Function]struct{}{
+	FUNCTION_RESETS:   {},
+	FUNCTION_CHANGES:  {},
+	FUNCTION_RATE:     {},
+	FUNCTION_INCREASE: {},
+	FUNCTION_DELTA:    {},
+}
+
+var FunctionsSupportingSmoothedMatrix = map[Function]struct{}{
+	FUNCTION_RATE:     {},
+	FUNCTION_INCREASE: {},
+	FUNCTION_DELTA:    {},
+}
```

Comment:
Can we derive these from `promql.AnchoredSafeFunctions` and `promql.SmoothedSafeFunctions`, rather than defining them ourselves?